src/libsyntax/parse/parser.rs

```diff
@@ -4017,8 +4032,16 @@ impl<'a> Parser<'a> {
                 None => {
                     let unused_attrs = |attrs: &[_], s: &mut Self| {
                         if attrs.len() > 0 {
-                            s.span_err(s.span,
-                                "expected statement after outer attribute");
+                            let unexpected_statement = s.span_to_snippet(s.last_span);
+                            if s.is_doc_comment(unexpected_statement) {
+                                s.span_err_help(s.last_span,
+                                    "found a documentation comment that doesn't document anything",
+                                    "The documentation string must be *before* the element being \
+                                    documented. Maybe you wanted a comment (starting with `//`)?")
```

Comment:
Stylistically I believe that compiler messages start with lowercase letters and don't contain punctuation, for example this could perhaps be:

> doc comments must come before what they document, maybe a comment was intended with `//`?